connectors/src/connectors/github/temporal/activities.ts

```
@@ -309,24 +310,36 @@ export async function githubUpsertIssueActivity(
     getIssuesInternalId(repoId),
     getRepositoryInternalId(repoId),
   ];
-  // TODO: last commentor, last comment date, issue labels (as tags)
-  await upsertDataSourceDocument({
-    dataSourceConfig,
-    documentId,
-    documentContent: renderedIssue,
-    documentUrl: issue.url,
-    timestampMs: updatedAtTimestamp,
-    tags: tags,
-    parents,
-    parentId: parents[1],
-    loggerArgs: logger.bindings(),
-    upsertContext: {
-      sync_type: isBatchSync ? "batch" : "incremental",
-    },
-    title: issue.title,
-    mimeType: MIME_TYPES.GITHUB.ISSUE,
-    async: true,
-  });
+
+  try {
+    // TODO: last commentor, last comment date, issue labels (as tags)
+    await upsertDataSourceDocument({
+      dataSourceConfig,
+      documentId,
+      documentContent: renderedIssue,
+      documentUrl: issue.url,
+      timestampMs: updatedAtTimestamp,
+      tags: tags,
+      parents,
+      parentId: parents[1],
+      loggerArgs: logger.bindings(),
+      upsertContext: {
+        sync_type: isBatchSync ? "batch" : "incremental",
+      },
+      title: issue.title,
+      mimeType: MIME_TYPES.GITHUB.ISSUE,
+      async: true,
+    });
+  } catch (e) {
+    if (axios.isAxiosError(e) && e.status === 403) {
+      // TODO be more precise about the error, check this
+      // https://app.datadoghq.eu/logs?query=%22Error%20uploading%20document%20to%20Dust%22&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host,service&event=AwAAAZV1BxPNAr2TXgAAABhBWlYxQnlDbEFBREFBNnp6aUt3S0JnQlAAAAAkMDE5NTc1MDgtODU3Ny00NjM3LWFjN2MtNWExYzNlZTgxMTBkAAARjw&fromUser=true&messageDisplay=inline&storage=hot&stream_sort=desc&viz=stream&from_ts=1741424278000&to_ts=1741425178000&live=false
+      // For now, just skip because it's probably a rate limit error
```

Not a rate limit error. This error is ours. Document is more than the DS document size limit

We will likely handle these errors by pausing the connector in the close future so I'm a bit antsy of keeping this code because it will silently ignore and the work we'll put in handling data source limits being reached will not pause the connector if that logic lives at a different layer.

The procedure so far has been to temporarily bump the plan limit or adjust the connector code to ignore upserts that are really big. Here I would test the size of documentContent and enforce a max size in github that is reasonable and places us below the 2mb limit.

---

Sorry, I was in a hurry, I know it's a document size (cursor suggested the comment and I accepted)

---

> The procedure so far has been to temporarily bump the plan limit or adjust the connector code to ignore upserts that are really big. Here I would test the size of documentContent and enforce a max size in github that is reasonable and places us below the 2mb limit.

We already have a restriction on size at 5mb (that I added recently and you approved :p)